File: src/community/script/py/src/test/resources/org/geoserver/script/py/buffer.py
from geoserver.wps import process
from geoscript.geom import Geometry

@process(
  inputs={'geom': (Geometry, 'The geometry to buffer'), 'distance':(float,'The buffer distance')}, 
  outputs={'result': (Geometry, 'The buffered geometry')}, 
  title='Buffer', 
  description='Buffers a geometry')
def run(geom, distance):
  return geom.buffer(distance);

File: src/community/script/rb/src/test/resources/org/geoserver/script/rb/buffer.rb
require 'java'

$title = 'Buffer'
$description = 'Buffers a geometry'

$inputs = { 
  "geom" => { 
    "title" => "The geometry to buffer", 
    "type" => org.locationtech.jts.geom.Geometry.java_class
   },
  "distance" => { 
    "title" => "The buffer distance", 
    "type" => java.lang.Double.java_class
  }
} 

$outputs = {
  "result" => {
    "title" => "The buffered geometry", 
    "type" => org.locationtech.jts.geom.Geometry.java_class
  }
}

def run(inputs)
  return { "result" => inputs["geom"].buffer(inputs["distance"]) }
  
end